cmd/gazelle/update-repos.go

```diff
@@ -32,9 +32,39 @@ import (
 type updateReposFn func(c *updateReposConfig, oldFile *rule.File, kinds map[string]rule.KindInfo) error
 
 type updateReposConfig struct {
-	fn           updateReposFn
-	lockFilename string
-	importPaths  []string
+	fn                      updateReposFn
+	lockFilename            string
+	importPaths             []string
+	buildExternalAttr       string
+	buildFileNamesAttr      string
+	buildFileGenerationAttr string
+	buildTagsAttr           string
+	buildFileProtoModeAttr  string
+	buildExtraArgsAttr      string
+}
+
+func checkValidBuildExternalAttr(attr string) bool {
```

Comment:
Could you extract these functions into an implementation of the `flag.Value` interface and move into the `//flag` package? Something like this would be good:

```
type AllowedStringFlag struct {
  Allowed []string
  Value *string
}

func (f *AllowedStringFlag) Set(v string) error {
  for _, a := range allowed {
    if v == a {
      *f.Value = v
      return nil
    }
  }
  return fmt.Errorf("invalid argument %q. Possible values are: %s", v, strings.Join(f.Allowed, ", "))
}
```